### PR TITLE
Fetch non-numeric track indices from MusicBrainz and Discogs

### PR DESCRIPTION
**Pull request by lengtche**
This is a PR in reference to issue #1831.

### CODE REVIEW

**Comment by sampsyo** [comment]
Great; thanks for getting the ball rolling here!

For the name of the field, maybe `alt_track` would suffice? Or, alternatively, `track_alt` to make the relationship with the current `track` field clear?

Also, to finish things up, would you mind:
* writing a short changelog entry that explains what's going on with that field
* cleaning up the whitespace changes
* taking a look at the [tests](https://travis-ci.org/beetbox/beets/jobs/188293344#L874), which look like they need updating

File: beets/autotag/mb.py [hunk 1]
```diff
@@ -265,6 +265,7 @@ def album_info(release):
             )
             ti.disctitle = disctitle
             ti.media = format
+            ti.alt_track_no = track['number']
```
Comment (sampsyo) [inline beets/autotag/mb.py]: To complete my understanding, what does this `number` field usually hold? Is it typically a number, but sometimes something else?

Comment (lengtche) [inline beets/autotag/mb.py]: My understanding from looking at a handful of XML dumps is that `position` always holds an integer index of the track and the `number` holds whatever the user inputted when adding tracks. If you had an album on a double-sided CD, tape, vinyl, etc., then there's the potential of having the `number` being something like `A1`, or whatever the case may be.

So it seems to me that simply adding this data should have zero impact with sorting.

**Comment by lengtche** [comment]
I can do what you requested.

I also like the idea of naming it `track_alt` for consistency.

File: beets/autotag/mb.py (hunk 1 above)
Comment (sampsyo) [inline beets/autotag/mb.py]: Cool; that helps! Thanks for elaborating.

**Comment by lengtche** [comment]
I believe I got everything done.

**Comment by sampsyo** [comment]
Looks great; thanks!

I think the problem is related to adding a fixed attribute for the field, which isn't really necessary—I'm going to flip this back to a flexible attribute and see how things go.

**Comment by sampsyo** [comment]
Looks like that worked! Thanks again for all your work; I'm merging now.

**Pull request merged**